src/cswinrt/strings/additions/Windows.Foundation/ExceptionDispatchHelper.cs

```diff
@@ -6,19 +6,31 @@
 namespace System.Threading.Tasks
 {
     using System;
-    using System.Runtime.ExceptionServices;
+    using System.Runtime.ExceptionServices;
+    using System.Runtime.InteropServices;
     using System.Threading;
     internal static class ExceptionDispatchHelper
-    {
+    {
+        private const int RPC_E_DISCONNECTED = unchecked((int)0x80010108);
+        private const int RPC_S_SERVER_UNAVAILABLE = unchecked((int)0x800706BA);
+        private const int JSCRIPT_E_CANTEXECUTE = unchecked((int)0x89020001);
+
         internal static void ThrowAsync(Exception exception, SynchronizationContext targetContext)
         {
             if (exception == null)
                 return;
 
-            // TODO - decide how to cleanly do it so it lights up if TA is defined
-            //if (exception is ThreadAbortException)
-            //    return;
-
+            // If this is an error indicating the RPC called failed,
+            // it is most likely due to the other process is gone.
+            // We ignore it as it just means we weren't able to report
+            // the completion and not actually an error in the other process.
+            if (exception is COMException comException &&
```

Comment:
Are we sure we'll always be getting a `COMException` in these cases? In the past there were some cases where CoreCLR and .NET Native would throw different exception types for the same HRESULT. Is this no longer a concern now between CoreCLR and Native AOT?

Reply:
We are doing the exception mapping in these invoke scenarios via our own function.  We aren't using the Marshal function.

Reply:
And right now, we don't map any of these.  If that changes, we can update it here too.